File: Selection Sort/SelectionSort-Java/SelectionSort.java
public class SelectionSort {
    public static int[] selectionSort(int[] arr) {
        for (int i = 0; i < arr.length; i++) {

            // find the index of the smallest value
            int min_index = i;
            for (int j = i + 1; j < arr.length; j++) {
                // if values after i are smaller, replace min_index
                if (arr[j] < arr[min_index]) {
                    min_index = j;
                }
            }

            // swap the smallest value with number at i
            int temp = arr[min_index];
            arr[min_index] = arr[i];
            arr[i] = temp;
        }

        return arr;
    }
}

File: Selection Sort/SelectionSort-Matlab/selectionSort.m
function array = selectionSort(array)
    arraySize = numel(array);
    for i = (1:arraySize-1)
        min = array(i);
        minIndex = i;
        
        for j = (i:arraySize)
            if array(j) <= min
                min = array(j);
                minIndex = j;
            end
        end
        
        if i ~= minIndex
            array([minIndex i]) = array([i minIndex]);
        end
    end         
end

